### ISSUE TITLE
Stable Diffusion pipeline reproducibility issue

### ISSUE THREAD

**Issue by 19and99** [huggingface/diffusers#3061]
### Describe the bug

Making `StableDiffusionPipeline` inference two times in a row with fixed seed gives different results.
The source of difference seems to come from the `AttnProcessor2_0` class, as the issue dissapears when using `CrossAttnProcessor` instead of  `AttnProcessor2_0`.

### Reproduction

```python
import torch
from diffusers import DDIMScheduler, StableDiffusionPipeline


model_id = "runwayml/stable-diffusion-v1-5"
pipe = StableDiffusionPipeline.from_pretrained(model_id).to("cuda")
pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config)
g = torch.Generator(device="cuda")

prompt = "A bear is playing a guitar on Times Square"

g.manual_seed(0)
result1 = pipe(prompt=prompt,
               num_inference_steps=50,
               generator=g,
               output_type='latent').images

g.manual_seed(0)
result2 = pipe(prompt=prompt,
               num_inference_steps=50,
               generator=g,
               output_type='latent').images

print("L_inf dist = ", abs(result1 - result2).max())
```

### System Info

- `diffusers` version: 0.15.0.dev0
- Platform: Linux-5.15.0-69-generic-x86_64-with-glibc2.31
- Python version: 3.9.16
- PyTorch version (GPU?): 2.0.0+cu117 (True)
- Huggingface_hub version: 0.13.3
- Transformers version: 4.27.4
- Accelerate version: 0.18.0
- xFormers version: not installed
- Using GPU in script?: Yes
- Using distributed or parallel set-up in script?: No

**Comment by patrickvonplaten** [comment]
Hey @19and99,

Yes this is actually expected since the new Attention2_0 makes use of randomness for matrix multiplication see: https://pytorch.org/docs/stable/notes/randomness.html

**Comment by patrickvonplaten** [comment]
```py
import os
os.environ["CUBLAS_WORKSPACE_CONFIG"] = ":16:8"

import torch
from diffusers import DDIMScheduler, StableDiffusionPipeline
import numpy as np

model_id = "runwayml/stable-diffusion-v1-5"
pipe = StableDiffusionPipeline.from_pretrained(model_id).to("cuda")
pipe.scheduler = DDIMScheduler.from_config(pipe.scheduler.config)
g = torch.Generator(device="cuda")

torch.backends.cudnn.benchmark = False
torch.use_deterministic_algorithms(True)

prompt = "A bear is playing a guitar on Times Square"

g.manual_seed(0)
result1 = pipe(prompt=prompt,
               num_inference_steps=50,
               generator=g,
               output_type='latent').images

g.manual_seed(0)
result2 = pipe(prompt=prompt,
               num_inference_steps=50,
               generator=g,
               output_type='latent').images

print("L_inf dist = ", abs(result1 - result2).max())
```
gives:
```
L_inf dist =  tensor(0., device='cuda:0')
```

**Comment by stevhliu** [comment]
Should we update the docs [here](https://huggingface.co/docs/diffusers/v0.15.0/en/using-diffusers/reproducibility#gpu) with a mention of how to force PyTorch to use deterministic algos @patrickvonplaten?
